src/Plugins/LevelDBStore/IO/Data/LevelDB/Iterator.cs

```diff
@@ -10,77 +10,131 @@
 // modifications are permitted.
 
 using System;
+using System.Runtime.InteropServices;
 
-namespace Neo.IO.Data.LevelDB
+namespace Neo.IO.Storage.LevelDB
 {
-    public class Iterator : IDisposable
+    /// <summary>
+    /// An iterator yields a sequence of key/value pairs from a database.
+    /// </summary>
+    public class Iterator : LevelDBHandle
     {
-        private IntPtr handle;
+        internal Iterator(nint handle)
+        {
+            Handle = handle;
+        }
 
-        internal Iterator(IntPtr handle)
+        /// <summary>
+        /// An iterator is either positioned at a key/value pair, or
+        /// not valid.  
+        /// </summary>
+        /// <returns>This method returns true if the iterator is valid.</returns>
+        public bool IsValid()
         {
-            this.handle = handle;
+            return Native.leveldb_iter_valid(Handle);
         }
 
-        private void CheckError()
+        /// <summary>
+        /// Position at the first key in the source.  
+        /// The iterator is Valid() after this call if the source is not empty.
+        /// </summary>
+        public void SeekToFirst()
         {
-            Native.leveldb_iter_get_error(handle, out IntPtr error);
-            NativeHelper.CheckError(error);
```

Comment:
Does we still need `NativeHelper`?

Reply:
When I get to a computer sometime this week. I can check

Reply:
I think it was removed now

Reply:
Yes ites removed and uses this now

```csharp
void Throw(Func<string, Exception> exception)
{
    Native.leveldb_iter_get_error(Handle, out var error);
    if (error != IntPtr.Zero) throw exception(Marshal.PtrToStringAnsi(error));
}
```